docs/sandwich-tests/index.md

```diff
@@ -0,0 +1,478 @@
+  is a required field of the `rule` stanza and is used to run commands.
+  - The
+    [`run`](https://dune.readthedocs.io/en/stable/reference/actions/run.html)
+    field is used to execute a program. In this case, it runs `node` on the
+    `SandwichTests.mjs` file and `time`s how long it takes.
+  - The default working directory for a rule is the mirrored directory in
+    `_build` that corresponds to the directory where the rule is located. Since
+    this rule is defined in `src/order-confirmation/dune`, the working directory
+    for it is `_build/default/src/order-confirmation`. Therefore the full path
+    of the file is
+
+    ```
+    _build/default/src/order-confirmation/output/src/order-confirmation/SandwichTests.mjs
+    ```
+
+Run `npm run test` to see your test being run through Dune.
+
+::: tip
+
+Rules are the true building blocks of Dune. A stanza like `melange.emit` is
+really just shorthand for a series of related rules. See [The Dune Mental
+Model](https://dune.readthedocs.io/en/stable/explanation/mental-model.html) for
+a good explanation of this.
+
+:::
+
+## `deps` field
+
+Change `SandwichTests.re` so that the test fails:
+
+<<< SandwichTests.re#broken-test
+
+Run `npm run test` again and... nothing happens. Dune won't re-run the test
+because we didn't add any dependencies to our rule. We need to add a
+`deps` field:
+
+```clj{3-5,7}
+(rule
+ (alias runtest)
+ (deps
+  Item.re
+  (:input ./output/src/order-confirmation/SandwichTests.mjs))
+ (action
+  (run time node %{input})))
+```
+
+This `deps` field has two values:
+
+- `Item.re` makes the rule depend on the `Item.re` file. This means that if
+  `Item.re` changes after the last successful test run, then running `npm run
+  test` will cause this test rule to run its `action` again.
+- `(:input ./output/src/order-confirmation/SandwichTests.mjs)` defines a new
+  [variable](https://dune.readthedocs.io/en/stable/concepts/variables.html#variables)
+  named `input` and binds it to the path
+  `./output/src/order-confirmation/SandwichTests.mjs`.
+
+Inside the action, we can reference the variable using `%{input}` (the name of
+the variable `input` enclosed by `%{` and `}`).
+
+See [Dependency
+Specification](https://dune.readthedocs.io/en/stable/concepts/dependency-spec.html)
+for the full lowdown on dependencies in Dune.
+
+We didn't have to use a variable here, but the alternative would've been to
+write the same path two times:
+
+```clj{5,7}
+(rule
+ (alias runtest)
+ (deps
+  Item.re
+  ./output/src/order-confirmation/SandwichTests.mjs)
+ (action
+  (run time node ./output/src/order-confirmation/SandwichTests.mjs)))
+```
+
+Run `npm run test` to see that your test runs. It fails, but you can fix it and
+run `npm run test` again. After the test succeeds, running `npm run test` won't
+cause any tests to run until either `Item.re` or `SandwichTests.mjs` is
+changed.
+
+---
+
+Subarashii! You now have a module for testing sandwich-related logic. In the
```

Comment:
This isn't Madame Jellobutter speaking. There are 3 characters in this book: (1) Madame Jellobutter, (2) "You", an unnamed coder who is a proxy for the reader, and (3) the unnamed omniscient narrator, who I like to call the Melange Fairy. The Melange Fairy sits on "Your" shoulder and tells them everything they need to know about Melange and naturally is fluent in all human languages.

The backstory of the Melange Fairy is that they are an advanced AI sent from the future to save humanity and teaching "you" Melange is the first step in that plan. Maybe this will be slowly explained over the course of the book, or maybe it won't be.